src/crypto/algorithms/megolm.ts

```diff
@@ -287,26 +300,28 @@ class MegolmEncryption extends EncryptionAlgorithm {
 
             try {
                 await this.shareSession(devicesInRoom, sharedHistory, singleOlmCreationPhase, blocked, session);
+                return session;
             } catch (e) {
                 logger.error(`Failed to ensure outbound session in ${this.roomId}`, e);
```

Comment:
Given that the error will be logged in the `catch` handler later, this seems redundant.

Reply:
Removed in https://github.com/matrix-org/matrix-js-sdk/pull/2908/commits/d5074776f117afe2af16756bfb74a3c9063740da.